wanted_attribute_builder.go

```diff
@@ -6,6 +6,11 @@ import (
 
 const ()
 
+type constraintInterface interface {
+	MarshalJSON() ([]byte, error)
+	isConstraint() bool // This function is not used but makes inheritance explicit
```

Comment:
Field rather than function?

Reply:
No. It's a function/method (technically Go calls them 'receivers' rather than methods) declaration.

source_constraint.go:83 has the definition (for SourceConstraint)

Reply:
Ah yes, sorry